raiden_libs/transport/matrix.py

```diff
@@ -73,7 +73,7 @@ def dispatch(self, room, event):
                 self.run_message_callbacks(event['content']['body'])
                 log.debug("{0}: {1}".format(event['sender'], event['content']['body']))
 
-    def transmit_data(self, message: str, target_node: str = None):
+    def transmit_data(self, message: str, target_node: str = None):  # type: ignore
```

Comment:
why dis `type: ignore` needed here.

Reply:
That doesn't make sense. Then you will get errors when you call the `send_message` function on the `MatrixTransport`.

Reply:
Nope, default in transport is none, so no problem when calling it. The problem is only non-matching signatures for type-checking. One could also add a `source_node = None` to matrix, but that wouldn't make sense, as in matrix logic the source node is known.

Reply:
This simplified script gives me the following error:
```py

class Transport:
    def send_message(
            self,
            message: str,
            source_node: str = None,
            target_node: str = None
    ):
        self.transmit_data(message, source_node, target_node)

    def transmit_data(self, message: str, source_node: str = None, target_node: str = None):
        raise NotImplementedError

class TestTransport(Transport):
    def transmit_data(self, message: str, target_node: str):
        print('message', message)
        print('target', target_node)


a = TestTransport()
a.send_message('message', 'from', 'to')
```

```
% python3 derive.py                                                                                                                                                                  10:55:48
Traceback (most recent call last):
  File "derive.py", line 21, in <module>
    a.send_message('message', 'from', 'to')
  File "derive.py", line 9, in send_message
    self.transmit_data(message, source_node, target_node)
TypeError: transmit_data() takes 3 positional arguments but 4 were given
```

So from my understanding this doesn't work.

Reply:
Okay I see, I'll push the changes to DummyTransport than instead so we don't have to change anything in Matrix.